src/appindexing.js

```diff
@@ -0,0 +1,130 @@
+	var numLinkTags = linkTags.length;
+
+	if (!numLinkTags) { return; }
+
+	for (var counter = 0; counter < numLinkTags; counter++) {
+
+		var currLinkTag = linkTags[counter];
+		var currLinkTagHref = currLinkTag.href;
+
+		if (!currLinkTagHref) { continue; }
+
+		if (currLinkTagHref.includes('ios-app')) {
+			appindexing.state['iOSAppIndexingTagsPresent'] = true;
+			currLinkTag.setAttribute('href',  appindexing.addBranchTrackingParams(currLinkTagHref));
+		}
+		if (currLinkTagHref.includes('android-app')) {
+			appindexing.state['androidAppIndexingTagsPresent'] = true;
+			currLinkTag.setAttribute('href',  appindexing.addBranchTrackingParams(currLinkTagHref));
+		}
+	}
+};
+
+/** Checks whether config contains appropriate per platform params. For Android, config needs to contain
+  * both androidPackageName and androidURL. For iOS both, iosAppId and iosURL. Once config is correctly set for
+  * iOS or Android, an App Indexing tag is built and inserted into the webpage.
+  * (Stage 2)
+  */
+appindexing.insertAppIndexingTagsFromConfig = function(options) {
+	if (options.hasOwnProperty('androidPackageName') &&
+		options.hasOwnProperty('androidURL') &&
+		typeof options['androidPackageName'] !== undefined &&
+		typeof options['androidURL'] !== undefined) {
+
+		appindexing.state['androidDetailsComplete'] = true;
+		appindexing.addAppIndexingTag('android', options);
+	}
+	if (options.hasOwnProperty('iosAppId') && 
+		options.hasOwnProperty('iosURL') && 
+		typeof options['iosAppId'] !== undefined && 
+		typeof options['iosURL'] !== undefined) {
+
+		appindexing.state['iOSDetailsComplete'] = true;
+		appindexing.addAppIndexingTag('iOS', options);
+	}
+};
+
+/** If App Indexing tags do not exist and if the original config passed into autoAppIndex call does 
+  * not contain appropriate data, then as a last resort, this function will try to update the config from page's App Links tags.
+  * Once config is built it is validated and then if appropriate, App Indexing tags are built and inserted into the webpage.
+  * (Stage 3)
+ */
+appindexing.populateConfigFromAppLinksTags = function(type, options) {
+	var metaTags = document.getElementsByTagName('meta');
+
+	for (var counter = 0; counter < metaTags.length; counter++) {
+		if (type=='iOS' && metaTags[counter].getAttribute('property') === 'al:ios:app_store_id') {
+			options['iosAppId'] = metaTags[counter].getAttribute('content');
+		}
+		if (type=='iOS' && metaTags[counter].getAttribute('property') === 'al:ios:url') {
+			options['iosURL'] = metaTags[counter].getAttribute('content').replace('://','/');
+		}
+		if (type=='android' && metaTags[counter].getAttribute('property') === 'al:android:package') {
+			options['androidPackageName'] = metaTags[counter].getAttribute('content');
+		}
+		if (type=='android' && metaTags[counter].getAttribute('property') === 'al:android:url') {
+			options['androidURL'] = metaTags[counter].getAttribute('content').replace('://','/');
+		}
+	}
```

Comment:
I would use a forEach here, it makes it a little cleaner. Either that or store metaTags[counter] since you access it a lot